base/inference.jl

```diff
@@ -1145,6 +1145,8 @@ function abstract_eval(e::ANY, vtypes::VarTable, sv::InferenceState)
         t = abstract_eval(e.args[1], vtypes, sv)
     elseif is(e.head,:inert)
         return abstract_eval_constant(e.args[1])
+    elseif is(e.head,:invoke)
+        error("type inference data-flow error: tried to double infer a function")
```

Comment:
Fair enough, but in the future we'll want to be able to re-infer functions to optimize them further.